src/colorprint.cc

```diff
@@ -73,13 +74,50 @@ PlatformColorCode GetPlatformColorCode(LogColor color) {
   };
 #endif
 }
+
+// Returns true iff Benchmark should use colors in the output.
+bool ShouldUseColor(bool stdout_is_tty) {
+  const char* const color_print = FLAG(color_print).c_str();
```

Comment:
It seems like it would be a lot easier to compare a std::string than a C string.

```c++
std::string const& color_print = FLAG(color_print);
const bool term_supports_color =
   term == "xterm" ||
   term == "foo" ||
  ...;
```